ceph.spec.in

```diff
@@ -1263,7 +1263,7 @@ ln -sf %{_sbindir}/mount.ceph %{buildroot}/sbin/mount.ceph
 install -m 0644 -D udev/50-rbd.rules %{buildroot}%{_udevrulesdir}/50-rbd.rules
 
 # sudoers.d
-install -m 0600 -D sudoers.d/ceph-osd-smartctl %{buildroot}%{_sysconfdir}/sudoers.d/ceph-osd-smartctl
+install -m 0440 -D sudoers.d/ceph-osd-smartctl %{buildroot}%{_sysconfdir}/sudoers.d/ceph-osd-smartctl
```

Comment:
Does this really fix the issue? My impression was/is that the permissions are determined by the line in the `%files` section, not by this line in the `%install` section.

Maybe prepend `%attr(440,root,root)` to the line in the `%files` section?

Reply:
I haven't tested this. I don't have a setup to build and test a build anywhere. I was taking a stab in the dark on this.

Reply:
The systems that are seeing the 600 permissions are CentOS and they're failing `visudo -c`. I just tested a Debian and Ubuntu host and they both have 440 and `visudo -c` succeeds on them.

Reply:
i just pushed this PR to ceph-ci to verify the fix. see https://shaman.ceph.com/builds/ceph/wip-38013-pr/1de14c7f9bcaf94152c62b50c37f4a44b6445f58/default/238891/

Reply:
```
$ wget 'https://3.chacra.ceph.com/r/ceph/wip-38013-pr/1de14c7f9bcaf94152c62b50c37f4a44b6445f58/centos/8/flavors/default/x86_64/ceph-osd-16.0.0-7069.g1de14c7f.el8.x86_64.rpm'
$ rpm2cpio  ceph-osd-16.0.0-7069.g1de14c7f.el8.x86_64.rpm | cpio -idmv
./etc/sudoers.d/ceph-osd-smartctl
...
$  ls -l ./etc/sudoers.d/ceph-osd-smartctl
-r--r----- 1 kefu kefu 199 Nov 12 14:50 ./etc/sudoers.d/ceph-osd-smartctl
```

Reply:
@smithfarm could you help take another look? i think this change does address the issue.

Reply:
Yup, in http://ftp.rpm.org/max-rpm/s1-rpm-anywhere-specifying-file-attributes.html it says:

"In cases where the package builder cannot create the files to be packaged with the proper ownership and permissions, the %attr macro can be used to make things right."

I guess the `%attr` is there mainly to enable the packager to override file permissions in an upstream release tarball.